### ISSUE TITLE
Not able to run ROS sample in M210 - Failed to read from serial device

### ISSUE THREAD

**Issue by imkhairulikhwan** [dji-sdk/Onboard-SDK-ROS#197]
Hi and good day, 

regarding the above issue, I have tried to check the forums and documentation but it seems all the settings are already correct. I have verified app Id, app key and baudrate as well in the configuration file (sdk.launch file). 

Baud rate that I use is 230400, which is found in DJI Assistant 2. I am connected to the drone using USB and USB-TTL cables.

I was able to run Qt sample in Win10 as well using the same setting and configuration. 

By the way, serial_name parameter is being set as default which is /dev/ttyUSB0. Could that be the reason? 

I attached here the screenshots to assist the troubleshooting.

Thank you in advance! 

![dji ros sample - failed to read from serial device](https://user-images.githubusercontent.com/4214164/39226050-8be5a6c4-4882-11e8-8252-3bc0f85fe66e.png)
![dji sdk linux ros - to check connected usb path](https://user-images.githubusercontent.com/4214164/39226051-8c27504c-4882-11e8-92d7-0b6e3b2559a5.png)

**Comment by imkhairulikhwan** [comment]
@rohitsantdji Thanks for the reply. Noted on that, I will proceed to verify the connection. 

I have checked and confirmed that I have followed all the steps on the setup, including adding the user into dialout and adding udev file. 

Anyhow, I will do the testing again today and revert back for the results. 

Thank you!

**Comment by imkhairulikhwan** [comment]
Hi @rohitsantdji , I still face the same error message when running the ROS sample. However, I noticed that I got the new Summary information when executing "roslaunch dji_sdk sdk.launch", as per attachment. 

Also, I have verified that indeed the usb drive is ttyUSB0 after typing the command that you provided. I'm sure the configuration is correct as I was able to run the Linux sample. 

If it helps, I attached here as well the catkin_ws folder where I execute above roslaunch command. Do we need to put the UserConfig.txt file as well in this folder?

Thank you in advance! 

![dji linux ros - failed to read from serial device with summary](https://user-images.githubusercontent.com/4214164/39280030-67e58548-492f-11e8-927e-7ea3d539b550.png)
![dji linux ros catkin_ws folder](https://user-images.githubusercontent.com/4214164/39280031-6c6356b8-492f-11e8-940b-8e829d74bb16.png)

**Comment by imkhairulikhwan** [comment]
I found from the existing issues #176 that it seems by commenting the code in dji_sdk_node.cpp, the problem will be solved. I'm not sure if it's the correct way, but now I am able to start the node. 

Also, I noticed that I need to do activation again even though in the same machine (dual boot), I have already activated in Windows 10. 

Seems I'm able to run ROS samples now :) But, I face another two issues. Should I create in separate tickets?

1. Running Advanced Sensing samples, I noticed the other terminal created only shows "Waiting for image", even though the camera has already been attached. Is it due to this message I got when cmake the sdk folder? p/s: I'm able to ping the other node as per screenshot.

"-- Did not find ximgproc in OpenCV
-- Did not find viz3d in OpenCV
-- Did not find required libraries, stereo vision depth perception sample will not be compiled.
"

2. Running ROS demo_mobile_comm sample shows message "Mobile Data Commands mode entered. Use OSDK Mobile App to use this feature", but when I tried to send "Take Photo" from iOS Mobile OnBoard SDK app, nothing happened. The mobile app only showed "Sending" and that's it. 

Thank you again, and sorry for the lengthy messages! 

![dji ros advanced sensing roslaunch - waiting for image](https://user-images.githubusercontent.com/4214164/39283594-885b1b3c-4941-11e8-8534-c780a9126075.png)
![dji ros demo_mobile_comm no error shown but no response either](https://user-images.githubusercontent.com/4214164/39283595-888f8ba6-4941-11e8-82bd-fff5aa72f1d3.png)
![dji ros - able to ping the node](https://user-images.githubusercontent.com/4214164/39283597-89124b04-4941-11e8-807f-8ab998d9628b.png)
